core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -23,6 +26,11 @@
  */
 class PageRegular extends Frontend
 {
+	/**
+	 * @var ResponseContext
+	 */
+	protected $responseContext;
```

Comment:
could someone possible change the response context and this would unintentionally keep state in the object?

Reply:
I have to allow child classes access to it and there's no way for a readonly property in a child class. So yes, that's possible. But they can also override all sorts of other methods and cause their version of PageRegular to not work anymore so I don't see any problem here.

Reply:
no I didn't mean that, I mean if someone switched the request context in the request. Maybe it would be better to access a "fresh copy" of the request context in every method that uses it instead of keeping state.

Reply:
No. It's the controller. It sets and ends the state. If a child class wants to tamper with it, then that's the fault of the child class. Simple as that.